src/elements/Branching.cpp

```diff
@@ -567,15 +573,16 @@ bool Branching::flpl_new_branch(TNodePtr &branching_node, NodePtr &new_node,
             // x0 and x1 define the range of the distribution, and x is your
             // power-law distributed variate.
             double y = uniform_(*(rnd_engine).get());
-            int x_0  = 2;
-            int x_1  = branching_node->get_branch()->size() - 2;
-            // TODO check where this 2 come from.
+            int x_0  = latbranch_dist_;
+            int x_1  =
+                branching_node->get_branch()->get_length() - latbranch_dist_;
+            // TODO check if we should not make n user-defined.
             int n = 2;
             double a =
-                (powf(x_1, (n + 1)) - powf(x_0, (n + 1))) * y + powf(x_0, (n + 1));
-            int x = (int)powf(a, 1. / (n + 1));
+                (powf(x_1, (n + 1)) - powf(x_0, (n + 1)))*y + powf(x_0, (n + 1));
+            double branching_dist = powf(a, 1. / (n + 1));
 
-            branching_point = x;
+            branching_point = get_closest_point(branching_node, branching_dist);
 
             // lateral branching on the elected node through the NEURITE.
```

Comment:
Same here, I modified the algorithm with 'key and max' becauses it draws a lot of random numbers